analyzers/src/SonarAnalyzer.CSharp/Rules/ArgumentSpecifiedForCallerInfoParameter.cs

```diff
@@ -50,11 +50,20 @@ public sealed class ArgumentSpecifiedForCallerInfoParameter : SonarDiagnosticAna
                 if (new CSharpMethodParameterLookup((InvocationExpressionSyntax)c.Node, c.SemanticModel) is { MethodSymbol: { } } methodParameterLookup
                     && methodParameterLookup.GetAllArgumentParameterMappings() is { } argumentMappings)
                 {
-                    foreach (var argumentMapping in argumentMappings.Where(x => x.Symbol.GetAttributes(CallerInfoAttributesToReportOn).Any()))
+                    foreach (var argumentMapping in argumentMappings.Where(x =>
+                        x.Symbol.GetAttributes(CallerInfoAttributesToReportOn).Any()
+                        && !IsArgumentPassthroughOfParameter(c.SemanticModel, x.Node, x.Symbol)))
                     {
                         c.ReportIssue(Diagnostic.Create(Rule, argumentMapping.Node.GetLocation()));
                     }
                 }
             }, SyntaxKind.InvocationExpression);
+
+        private static bool IsArgumentPassthroughOfParameter(SemanticModel semanticModel, ArgumentSyntax argument, IParameterSymbol targetParameter) =>
+            semanticModel.GetSymbolInfo(argument.Expression).Symbol is IParameterSymbol sourceParameter // the argument passed to the method is itself an parameter.
+                                                                                                        // Let's check if it has the same attributes.
+                && sourceParameter.GetAttributes(CallerInfoAttributesToReportOn).ToList() is var sourceAttributes
+                && targetParameter.GetAttributes(CallerInfoAttributesToReportOn).ToList() is var targetAttributes
```

Comment:
It looks like we ask for the symbol attributes twice. 

What if we update the code like this?
```
foreach (var argumentMapping in argumentMappings.Where(x =>
                        x.Symbol.GetAttributes(CallerInfoAttributesToReportOn).ToList() is var targetAttributes
                        && targetAttributes.Any()
                        && !IsArgumentPassthroughOfParameter(c.SemanticModel, x.Node, targetAttributes)))
                    {
                        c.ReportIssue(Diagnostic.Create(Rule, argumentMapping.Node.GetLocation()));
                    }

        private static bool IsArgumentPassthroughOfParameter(SemanticModel semanticModel, ArgumentSyntax argument, List<AttributeData> targetAttributes) =>
            semanticModel.GetSymbolInfo(argument.Expression).Symbol is IParameterSymbol sourceParameter // the argument passed to the method is itself an parameter.
                                                                                                        // Let's check if it has the same attributes.
                && sourceParameter.GetAttributes(CallerInfoAttributesToReportOn).ToList() is var sourceAttributes
                && targetAttributes.All(target => sourceAttributes.Any(source => target.AttributeClass.Name == source.AttributeClass.Name));

```

Reply:
My reasoning was:

* In 99% of the cases there is no attribute on an invoked method parameter and even if in 99% of the cases none of the CallerInfo attributes are present.
* In these cases `Any()` is slightly cheaper in time and space.
* The double query of the attributes is performed only very very rarely.